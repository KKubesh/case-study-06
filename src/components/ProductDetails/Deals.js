import React, {Component} from 'react';
import tagIcon from '../../images/tag.png';

class Deals extends Component {
    render () {
        return (
            <div>
                <div className="deals" data-testid="deals">
                    <img src={tagIcon} alt="tag icon" /> spend $50, ship free<br/>
                    <img src={tagIcon} alt="tag icon" /> $25 gift card with purchase of a select Ninja Blender
                </div>
                <hr/>
            </div>
        )
    }
}

export default Deals;