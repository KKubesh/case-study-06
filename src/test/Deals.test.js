import React from 'react';
import { render, cleanup } from 'react-testing-library';
import Deals from '../components/ProductDetails/Deals';

afterEach(cleanup);

test('<Deals />', () => {
    const { getByTestId, container } = render(<Deals/>);
    expect(getByTestId('deals').textContent).toBe(" spend $50, ship free $25 gift card with purchase of a select Ninja Blender");
    expect(container.firstChild).toMatchSnapshot();
});