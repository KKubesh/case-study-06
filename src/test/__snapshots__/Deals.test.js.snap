// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Deals /> 1`] = `
<div>
  <div
    class="deals"
    data-testid="deals"
  >
    <img
      alt="tag icon"
      src="tag.png"
    />
     spend $50, ship free
    <br />
    <img
      alt="tag icon"
      src="tag.png"
    />
     $25 gift card with purchase of a select Ninja Blender
  </div>
  <hr />
</div>
`;
